ibc-core/ics02-client/context/src/context.rs

```diff
@@ -102,3 +103,53 @@ pub trait ClientExecutionContext:
         height: Height,
     ) -> Result<(), ContextError>;
 }
+
+/// Extends the client validation context capabilities by providing additional
+/// methods for validating a client state, particularly benefiting ICS-07
+/// Tendermint clients by granting access to essential information from hosts.
+///
+/// Categorized under ICS-02, it can also be utilized by other types of light
+/// clients. Developers can view this trait as a custom context definition
+/// example that expands validation capabilities, according to their specific
+/// light client requirements.
+pub trait ExtClientValidationContext: ClientValidationContext {
+    /// Returns the current timestamp of the local chain.
+    fn host_timestamp(&self) -> Result<Timestamp, ContextError>;
+
+    /// Returns the current height of the local chain.
+    fn host_height(&self) -> Result<Height, ContextError>;
+
+    /// Returns all the heights at which a consensus state is stored
+    fn consensus_state_heights(&self, client_id: &ClientId) -> Result<Vec<Height>, ContextError>;
+
+    /// Search for the lowest consensus state higher than `height`.
+    fn next_consensus_state(
+        &self,
+        client_id: &ClientId,
+        height: &Height,
+    ) -> Result<Option<Self::ConsensusStateRef>, ContextError>;
+
+    /// Search for the highest consensus state lower than `height`.
+    fn prev_consensus_state(
+        &self,
+        client_id: &ClientId,
+        height: &Height,
+    ) -> Result<Option<Self::ConsensusStateRef>, ContextError>;
+}
+
+/// Extends the client context required during execution.
+///
+/// This trait is automatically implemented for all types that implement
+/// [`ExtClientValidationContext`] and [`ClientExecutionContext`]
+pub trait ExtClientExecutionContext: ExtClientValidationContext + ClientExecutionContext {}
+
+impl<T> ExtClientExecutionContext for T where T: ExtClientValidationContext + ClientExecutionContext {}
```

Comment:
As far as I can tell, based off of how this trait is being used, it's essentially a trait alias for any types that implement `ExtClientValidationContext` and `ClientExecutionContext`. Is that an appropriate way to think about this?

Reply:
Right. Updated the docstring: [d4e66c2](https://github.com/cosmos/ibc-rs/pull/1166/commits/d4e66c2229b49ea9903b18bb628a9339a603883c)
Hope it's more clear now.